### PR TITLE
Add Perplexity metric

### PR DESCRIPTION
**Pull request by jeffcarp**

### CODE REVIEW

**Comment by jshin1394** [comment]
Seems like there is an error with `ERROR: Could not find a version that satisfies the requirement tensorflow==2.18.0 (from versions: none)
ERROR: No matching distribution found for tensorflow==2.18.0`

It is weird as I can see tensorflow 2.18.0 in their pypi page https://pypi.org/project/tensorflow/#history

File: src/metrax/metrics.py [hunk 1]
```diff
@@ -536,4 +536,76 @@ def compute(self) -> jax.Array:
...
+      predictions: jax.Array,
+      labels: jax.Array,
+      sample_weights: jax.Array | None = None,
+  ) -> 'Perplexity':
+    """Updates the metric.
+
+    Args:
+      predictions: A floating point 2D vector representing the prediction
+        generated from the model. The shape should be (batch_size, seq_len).
+      labels: True value. The shape should be (batch_size, seq_len).
```
Comment (jshin1394) [inline src/metrax/metrics.py:571]: does this mean that this can be multi label? for instance, predictions may be [5 0 2] and the label is [1 0 1]?

Comment (jshin1394) [reply src/metrax/metrics.py:571]: actually, getting the probability of each element inside predictions in line 583 suggests that we have a single label scenario

Comment (jeffcarp) [reply src/metrax/metrics.py:571]: Yeah I don't think this works for multi-label currently (it also converts the labels to one-hot on L586). How common are labels with >1 truth values in NLP eval?

Comment (jshin1394) [reply src/metrax/metrics.py:571]: then the shape of labels is (batch_size, ) then?

**Comment by jeffcarp** [comment]
> Seems like there is an error with `ERROR: Could not find a version that satisfies the requirement tensorflow==2.18.0 (from versions: none) ERROR: No matching distribution found for tensorflow==2.18.0`

Specifying Python 3.12 (one major version behind the latest) fixed the problem.
